### ISSUE TITLE
wrong gb over ZZ

### ISSUE THREAD

**Issue by jakobkroeker** [Macaulay2/M2#243]
```
rng = ZZ[x_1]
I = ideal(-6*x_1^3+10*x_1-10,-2*x_1^2-x_1,-x_1^3,-4*x_1^3+8*x_1^2)
gI = ideal gens gb I;
assert( ( (gens I)%gI) == 0 ); --fails
gI
```
output
```
i1 : rng = ZZ[x_1]

o1 = rng

o1 : PolynomialRing

i2 : I = ideal(-6*x_1^3+10*x_1-10,-2*x_1^2-x_1,-x_1^3,-4*x_1^3+8*x_1^2)

                3                  2         3      3     2
o2 = ideal (- 6x  + 10x  - 10, - 2x  - x , -x , - 4x  + 8x )
                1      1           1    1    1      1     1

o2 : Ideal of rng

i3 : gI = ideal gens gb I;

o3 : Ideal of rng

i4 : assert( ( (gens I)%gI) == 0 ); --fails
stdio:4:1:(3): error: assertion failed

i5 : gI

o5 = ideal (20, x )
                 1

o5 : Ideal of rng

```

I didn't check by hand, but I suppose, the correct answer is 
ideal (10, x )

**Comment by mikestillman** [comment]
Good, thanks Jakob, I’ll take a look.  I have been working on fixing a couple of bugs along these lines, and (previously!) I had one left.

**Comment by jakobkroeker** [comment]
simplified the example   
```
rng = ZZ[x_1]
I = ideal(-6*x_1^3-10,-2*x_1^2-x_1,-4*x_1^3)
gI = ideal gens gb I;
assert( ( (gens I)%gI) == 0 ); --fails
gI
```

**Comment by jakobkroeker** [comment]
>Good, thanks Jakob, I’ll take a look. 

let me know If the examples are simple enough,
If not, I will try to find smalller ones.

**Comment by mikestillman** [comment]
Fixed in commit 389dc05451a7d0b1703cdf8b9e660de76e4d791b.  These tests were incorporated into a test file test/gbZZ-2.m2
